jwst/cube_build/src/cube_match_sky.c

```diff
@@ -0,0 +1,1441 @@
+   1 set the DQ plane based on the FOV
+weight_type : int
+   0: use emsm weighting
+   1: use msm weighting
+start_region : int
+    starting slice number for detector region used in dq flagging
+end_region: int 
+    ending slice number for detector region used in dq flagging
+overlap_partial : int
+    a dq flag indicating that only a portion of the spaxel is overlapped by a mapped detector pixel
+overlap_full : int
+    a dq flag indicating that the entire spaxel is overlapped by the mapped detector pixel
+xcoord : double array
+   size of naxis1. This array holds the center x axis values of the ifu cube 
+ycoord : double array
+   size of naxis2. This array holds the center y axis values of the ifu cube 
+zcoord : double array
+   size of naxis3. This array holds the center x axis values of the ifu cube 
+flux : double array
+   size: point cloud elements. Flux of each point cloud memeber
+err : double array
+   size: point cloud elements. err of each point cloud memeber
+slice_no: int
+   slice number of point cloud member to be in dq flagging
+coord1 : double array
+   size: point cloud elements. Naxis 1 coordinate of point cloud member (xi) 
+coord2 : double array
+   size: point cloud elements. Naxis 2 coordinate of point cloud member (eta)
+wave : double array
+   size: point cloud elements. Wavelegnth of each point cloud memeber
+rois_pixel : double array
+   size: point cloud elements. Roi in spatial dimension to use for point cloud memeber
+roiw_pixel : double array
+   size: point cloud elements. Roi in wavelength dimension to use point cloud memeber
+scalerad_pixel : double array
+   size: point cloud elements. MSM weight parameter to use for point cloud member
+zcdelt3: double array
+   size: point cloud elements. Spectral scale to use at wavelength of point cloud member
+roiw_ave : double
+   Average roiw for all the wavelength planes. Used in dq flagging
+cdelt1 : double
+   Naxis 1 scale for cube
+cdelt2 : double
+   Naxis 2 scale for cube
+
+
+Returns
+-------
+spaxel_flux : numpy.ndarray
+  IFU spaxel cflux
+spaxel_weight : numpy.ndarray
+  IFU spaxel weight 
+spaxel_iflux : numpy.ndarray
+  IFU spaxel weight map (number of overlaps) 
+spaxel_var : numpy.ndarray
+  IFU spaxel error
+spaxel_dq : numpy.ndarray
+  IFU spaxel dq
+*/
+
+#include <stdlib.h>
+#include <math.h>
+#include <stdio.h>
+#include <Python.h>
+#include <stdbool.h>
+#include <numpy/arrayobject.h>
+#include <numpy/npy_math.h>
+
+#define PY_ARRAY_UNIQUE_SYMBOL _jwst_cube_match_sky_numpy_api    //WHAT IS THIS AND WHERE IS IT USED???
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+
+// routines used from cube_utils.c
+
+extern double sh_find_overlap(const double xcenter, const double ycenter, 
+		      const double xlength, const double ylength,
+			      double xPixelCorner[],double yPixelCorner[]);
+
+// Allocate the memory to for the spaxel arrays in the cube
+
+int mem_alloc(int nelem, double **fluxv, double **weightv, double **varv,  double **ifluxv) {
```

Comment:
Isn't this the same function as in `cube_match_internal.c`? Maybe it should be placed in a separate file and then included in both modules?